From c3596383dfb0e6b96ff34bfa8792c190e8b6d564 Mon Sep 17 00:00:00 2001
From: Thomas Bourgoin <thomas.bourgoin@foss.st.com>
Date: Mon, 11 Mar 2024 15:14:10 +0100
Subject: [PATCH 0777/1695] dts: stm32: disable IWDG1 and IWDG2 during low
 power on stm32mp135f-dk

Update shadow value of otp HW2 to disable IWDG1 and IWDG2 during standby
and stop on stm32mp135f-dk.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git a3e0c9900240bd7142cf351c440f8c3292d69e70 dts: stm32: disable IWDG1 and IWDG2 during low power on stm32mp135f-dk]
Signed-off-by: Thomas Bourgoin <thomas.bourgoin@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/366595
Change-Id: I8352bb03c1079136b880d0046a04a4ef531b58ca
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/372975
Reviewed-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Domain-Review: Yann GAUTIER <yann.gautier@foss.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
---
 core/arch/arm/dts/stm32mp135f-dk.dts | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/core/arch/arm/dts/stm32mp135f-dk.dts b/core/arch/arm/dts/stm32mp135f-dk.dts
index e569d24c1..60d3741e8 100644
--- a/core/arch/arm/dts/stm32mp135f-dk.dts
+++ b/core/arch/arm/dts/stm32mp135f-dk.dts
@@ -12,6 +12,7 @@
 #include <dt-bindings/mfd/st,stpmic1.h>
 #include <dt-bindings/power/stm32mp1-power.h>
 #include <dt-bindings/regulator/st,stm32mp13-regulator.h>
+#include <dt-bindings/soc/stm32mp-provisioning.h>
 #include <dt-bindings/soc/stm32mp-tzc400-macro.h>
 #include <dt-bindings/soc/stm32mp13-etzpc.h>
 #include <dt-bindings/soc/stm32mp13-etzpc.h>
@@ -58,6 +59,15 @@
 		};
 	};
 
+	shadow-prov {
+		compatible = "st,provisioning";
+		hw2_prov {
+			nvmem-cells = <&hw2_otp>;
+			st,shadow-value = <0x000001E0>;
+			st,shadow-lock = <STICKY_LOCK_SR>;
+		};
+	};
+
 	typec_wakeup: typec_wakeup {
 		compatible = "st,stm32mp1-pwr-irq-user";
 		wakeup-parent = <&pwr_irq>;
-- 
2.47.2

